extensions/amp-social-share/0.2/social-share.js

```diff
@@ -30,63 +32,196 @@ const DEFAULT_HEIGHT = 44;
  * @param {!JsonObject} props
  * @return {PreactDef.Renderable}
  */
-export function SocialShare(props) {
+export function SocialShare2(props) {
+  const name = 'SocialShare2';
+  useResourcesNotify();
+
+  const {
+    typeConfig,
+    baseEndpoint,
+    checkedWidth,
+    checkedHeight,
+    obsoleteType,
+  } = checkProps_(props, name);
+  if (obsoleteType) {
+    return;
+  }
+  const finalEndpoint = createEndpoint_(typeConfig, baseEndpoint, props);
+
   /**
-   * Handle key presses on the element.
-   * @param {!Event} event
    * @private
+   * @param {!JsonObject} props
+   * @param {?string} name
+   * @return {!JsonObject}
    */
-  function handleKeyPress(event) {
-    const {key} = event;
-    if (key == Keys.SPACE || key == Keys.ENTER) {
-      event.preventDefault();
-      handleActivation();
+  function checkProps_(props, name) {
```

Comment:
This looks like a static function? If so, let's make it a module-level function. Also, let's drop `_` for this and others.